addons/interact_menu/functions/fnc_removeActionFromClass.sqf

```diff
@@ -17,13 +18,46 @@
  * Public: No
  */
 
-params ["_objectType", "_typeNum", "_fullPath"];
+params ["_objectType", "_typeNum", "_fullPath", ["_useInheritance", false, [false]]];
 
 _objectType = _objectType call EFUNC(common,getConfigName);
 
 private _res = _fullPath call FUNC(splitPath);
 _res params ["_parentPath", "_actionName"];
 
+if (_useInheritance) exitWith {
+    // Only need to run for classes that have already been initialized
+    {
+        [_x, _typeNum, _fullPath] call FUNC(removeActionFromClass);
+    } forEach (GVAR(inheritedClassesAll) select {_x isKindOf _objectType});
+
+    // Find same path and actionName, and check if it's a parent class, needs to be checked for all classes
+    private _index = GVAR(inheritedActionsAll) findIf {
+        params ["_currentType", "", "_currentParentPath", "_currentAction"];
+
+        [_objectType isKindOf _currentType, _currentParentPath, _currentAction select 0] isEqualTo [true, _parentPath, _actionName]
+    };
+
+     // Add to exclude classes
+    if (_index != -1) then {
+        (GVAR(inheritedActionsAll) select _index select 4) pushBackUnique _objectType;
```

Comment:
I'm not a fan of this, as it means you can't readd an action that you have removed, even if it's the initial class you passed as `_objectType`.

Reply:
I was wrong, you can readd actions.